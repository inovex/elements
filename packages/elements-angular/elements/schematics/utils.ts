import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
