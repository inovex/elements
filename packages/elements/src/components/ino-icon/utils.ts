import { getAssetPath } from '@stencil/core';

let CACHED_MAP: Map<string, string>;

export function getIconMap(): Map<string, string> {
	if (!CACHED_MAP) {
		const win = window as any;
		win.inoIcons = win.inoIcons || {};
		CACHED_MAP = win.inoIcons.map = win.inoIcons.map || new Map();
	}
	return CACHED_MAP;
}

export function addIcons(icons: { [name: string]: string }) {
	const map = getIconMap();
	Object.keys(icons).forEach((name) => map.set(name, icons[name]));
}

export const getSrc = (src: string | undefined) => {
	if (isStr(src)) {
		src = src.trim();
		if (isSrc(src)) {
			return src;
		}
	}
	return null;
};

export function getName(name: string | undefined) {
	if (name && !isSrc(name)) {
		return name;
	}
	return undefined;
}

export function getUrl(src: string | undefined, icon: string | undefined) {
	let url = getSrc(src);

	if (url) {
		return url;
	}
	url = getName(icon);
	if (url) {
		return getNamedUrl(url);
	}

	url = getSrc(icon);
	if (url) {
		return url;
	}

	return null;
}

export function getNamedUrl(name: string) {
	const url = getIconMap().get(name);
	if (url) {
		return url;
	}
	return getAssetPath(`./ino-icon/${name}.svg`);
}

export const isSrc = (str: string) => str.length > 0 && /(\/|\.)/.test(str);

export const isStr = (val: any): val is string => typeof val === 'string';

export const toLower = (val: string) => val.toLowerCase();
