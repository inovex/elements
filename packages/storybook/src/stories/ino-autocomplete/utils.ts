import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
