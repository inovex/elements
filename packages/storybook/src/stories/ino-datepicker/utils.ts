import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
