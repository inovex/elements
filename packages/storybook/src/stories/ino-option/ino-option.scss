.story-ino-option {
  padding: 40px;

  ino-select {
    margin-bottom: 30px;
  }
}
