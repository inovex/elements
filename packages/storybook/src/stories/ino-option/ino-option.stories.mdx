import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-option/readme.md";

<Meta title="Input/ino-option/Documentation" />

<Description markdown={readme} />
