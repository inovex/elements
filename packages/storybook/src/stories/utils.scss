/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
