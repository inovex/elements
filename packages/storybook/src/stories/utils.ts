import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
