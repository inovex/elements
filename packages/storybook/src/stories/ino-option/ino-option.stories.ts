import { html } from 'lit-html';
import { defaultDecorator } from '../utils';
import './ino-option.scss';

export default {
  title: 'Input/ino-option',
  component: 'ino-option',
  decorators: [(story) => defaultDecorator(story, 'story-ino-option')],
};

export const Playground = (args) => html`
  <div class="story-option">
    <ino-select label="Customizable option">
      <ino-option
        class="customizable-option"
        disabled="${args.disabled}"
        value="${args.value}"
      >
        Content
      </ino-option>
      <ino-option
          class="customizable-option"
          disabled="${args.disabled}"
          value="${args.value}"
      >
        Content
      </ino-option>
    </ino-select>
  </div>
`;
Playground.args = {
  disabled: false,
  value: 'Content',
}

export const States = () => html`
  <div class="story-option">
    <ino-select label="Disabled option">
      <ino-option value="Content" disabled>Content</ino-option>
    </ino-select>

    <ino-select label="Selected option">
      <ino-option value="Content" selected>Content</ino-option>
    </ino-select>
  </div>
`;

// export const SelectOptions = () => /*html*/ `
//     <style>
//         ino-option.customizable-option {
//           --ino-option-selected-background-color: ${text(
//             '--ino-option-selected-background-color',
//             'rgba(61, 64, 245, 0.05)',
//             'Custom Properties'
//           )};
//           --ino-option-selected-background-color-hover: ${text(
//             '--ino-option-selected-background-color-hover',
//             'rgba(61, 64, 245, 0.1)',
//             'Custom Properties'
//           )};
//           --ino-option-selected-background-color-focus: ${text(
//             '--ino-option-selected-background-color-focus',
//             'rgba(61, 64, 245, 0.15)',
//             'Custom Properties'
//           )};
//           --ino-option-selected-background-color-active: ${text(
//             '--ino-option-selected-background-color-active',
//             'rgba(61, 64, 245, 0.3)',
//             'Custom Properties'
//           )};
//           --ino-option-deselected-background-color: ${text(
//             '--ino-option-deselected-background-color',
//             '',
//             'Custom Properties'
//           )};
//           --ino-option-deselected-background-color-hover: ${text(
//             '--ino-option-deselected-background-color-hover',
//             '',
//             'Custom Properties'
//           )};
//           --ino-option-deselected-background-color-focus: ${text(
//             '--ino-option-deselected-background-color-focus',
//             '',
//             'Custom Properties'
//           )};
//           --ino-option-deselected-background-color-active: ${text(
//             '--ino-option-deselected-background-color-active',
//             '',
//             'Custom Properties'
//           )};
//         }
//     </style>
