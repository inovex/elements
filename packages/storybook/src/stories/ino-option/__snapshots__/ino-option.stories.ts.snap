// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input/ino-option DisabledOption smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-option">
    <div class="story-option">
      <ino-select label="My Select"
                  class="hydrated"
      >
        <div class="mdc-select mdc-select--filled">
          <div class="mdc-select__anchor mdc-ripple-upgraded"
               tabindex="0"
               aria-disabled="false"
               aria-invalid="false"
          >
            <div class="mdc-select__selected-text">
              Content
            </div>
            <span class="mdc-select__dropdown-icon">
              <svg class="mdc-select__dropdown-icon-graphic"
                   viewbox="7 10 10 5"
              >
                <polygon class="mdc-select__dropdown-icon-inactive"
                         stroke="none"
                         fill-rule="evenodd"
                         points="7 10 12 15 17 10"
                >
                </polygon>
                <polygon class="mdc-select__dropdown-icon-active"
                         stroke="none"
                         fill-rule="evenodd"
                         points="7 15 12 10 17 15"
                >
                </polygon>
              </svg>
            </span>
            <ino-label class="hydrated">
              <div class="mdc-line-ripple">
              </div>
              <label class="mdc-floating-label mdc-floating-label--float-above">
                My Select
              </label>
            </ino-label>
          </div>
          <div class="mdc-select__menu mdc-menu mdc-menu-surface mdc-menu-surface--fullwidth">
            <ul class="mdc-deprecated-list">
              <ino-option class="customizable-option hydrated"
                          disabled="true"
                          selected="undefined"
                          value="some-id"
              >
                <li class="mdc-deprecated-list-item mdc-deprecated-list-item--disabled mdc-deprecated-list-item--selected"
                    data-value="some-id"
                    aria-selected
                    tabindex="-1"
                >
                  <span class="mdc-deprecated-list-item__text">
                    Content
                  </span>
                </li>
              </ino-option>
            </ul>
          </div>
        </div>
      </ino-select>
    </div>
  </div>
</div>
`;

exports[`Input/ino-option Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-option">
    <div class="story-option">
      <ino-select label="My Select"
                  class="hydrated"
      >
        <div class="mdc-select mdc-select--filled">
          <div class="mdc-select__anchor mdc-ripple-upgraded"
               tabindex="0"
               aria-disabled="false"
               aria-invalid="false"
          >
            <div class="mdc-select__selected-text">
            </div>
            <span class="mdc-select__dropdown-icon">
              <svg class="mdc-select__dropdown-icon-graphic"
                   viewbox="7 10 10 5"
              >
                <polygon class="mdc-select__dropdown-icon-inactive"
                         stroke="none"
                         fill-rule="evenodd"
                         points="7 10 12 15 17 10"
                >
                </polygon>
                <polygon class="mdc-select__dropdown-icon-active"
                         stroke="none"
                         fill-rule="evenodd"
                         points="7 15 12 10 17 15"
                >
                </polygon>
              </svg>
            </span>
            <ino-label class="hydrated">
              <div class="mdc-line-ripple">
              </div>
              <label class="mdc-floating-label">
                My Select
              </label>
            </ino-label>
          </div>
          <div class="mdc-select__menu mdc-menu mdc-menu-surface mdc-menu-surface--fullwidth">
            <ul class="mdc-deprecated-list">
              <ino-option class="customizable-option hydrated"
                          disabled="false"
                          selected="false"
                          value="some-id"
              >
                <li class="mdc-deprecated-list-item"
                    data-value="some-id"
                    tabindex="-1"
                >
                  <span class="mdc-deprecated-list-item__text">
                    Content
                  </span>
                </li>
              </ino-option>
            </ul>
          </div>
        </div>
      </ino-select>
    </div>
  </div>
</div>
`;

exports[`Input/ino-option SelectedOption smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-option">
    <div class="story-option">
      <ino-select label="My Select"
                  class="hydrated"
      >
        <div class="mdc-select mdc-select--filled">
          <div class="mdc-select__anchor mdc-ripple-upgraded"
               tabindex="0"
               aria-disabled="false"
               aria-invalid="false"
          >
            <div class="mdc-select__selected-text">
              Content
            </div>
            <span class="mdc-select__dropdown-icon">
              <svg class="mdc-select__dropdown-icon-graphic"
                   viewbox="7 10 10 5"
              >
                <polygon class="mdc-select__dropdown-icon-inactive"
                         stroke="none"
                         fill-rule="evenodd"
                         points="7 10 12 15 17 10"
                >
                </polygon>
                <polygon class="mdc-select__dropdown-icon-active"
                         stroke="none"
                         fill-rule="evenodd"
                         points="7 15 12 10 17 15"
                >
                </polygon>
              </svg>
            </span>
            <ino-label class="hydrated">
              <div class="mdc-line-ripple">
              </div>
              <label class="mdc-floating-label mdc-floating-label--float-above">
                My Select
              </label>
            </ino-label>
          </div>
          <div class="mdc-select__menu mdc-menu mdc-menu-surface mdc-menu-surface--fullwidth">
            <ul class="mdc-deprecated-list">
              <ino-option class="customizable-option hydrated"
                          disabled="undefined"
                          selected="true"
                          value="some-id"
              >
                <li class="mdc-deprecated-list-item mdc-deprecated-list-item--disabled mdc-deprecated-list-item--selected"
                    data-value="some-id"
                    aria-selected
                    tabindex="-1"
                >
                  <span class="mdc-deprecated-list-item__text">
                    Content
                  </span>
                </li>
              </ino-option>
            </ul>
          </div>
        </div>
      </ino-select>
    </div>
  </div>
</div>
`;
